src/es/lectortmo/src/eu/kanade/tachiyomi/extension/es/lectortmo/LectorTmo.kt

```diff
@@ -575,37 +616,137 @@ class LectorTmo : ParsedHttpSource(), ConfigurableSource {
 
     private fun getScanlatorPref(): Boolean = preferences.getBoolean(SCANLATOR_PREF, SCANLATOR_PREF_DEFAULT_VALUE)
 
-    private fun getSFWModePref(): Boolean = preferences.getBoolean(SFW_MODE_PREF, SFW_MODE_PREF_DEFAULT_VALUE)
-
     private fun getSaveLastCFUrlPref(): Boolean = preferences.getBoolean(SAVE_LAST_CF_URL_PREF, SAVE_LAST_CF_URL_PREF_DEFAULT_VALUE)
 
+    private fun checkBox(
+        ctx: Context,
+        key: String,
+        title: String,
+        summary: String? = null,
+        default: Boolean = false,
+    ) = CheckBoxPreference(ctx).apply {
+        this.key = key
+        this.title = title
+        summary?.let { this.summary = it }
+        setDefaultValue(default)
+    }
+
     override fun setupPreferenceScreen(screen: PreferenceScreen) {
-        val sfwModePref = CheckBoxPreference(screen.context).apply {
-            key = SFW_MODE_PREF
-            title = SFW_MODE_PREF_TITLE
-            summary = SFW_MODE_PREF_SUMMARY
-            setDefaultValue(SFW_MODE_PREF_DEFAULT_VALUE)
-        }
+        val ctx = screen.context
+
+        val nsfwGeneral = checkBox(
+            ctx,
+            SFW_GENERAL,
+            "Ocultar todo el contenido NSFW",
+            "Bloquea automáticamente Ecchi, GL, BL, Harem y Trap",
+        )
+
+        val ecchi = checkBox(ctx, NSFW_ECCHI, "    • Ocultar Ecchi")
+        val gl = checkBox(ctx, NSFW_GIRLS_LOVE, "    • Ocultar Girls Love")
+        val bl = checkBox(ctx, NSFW_BOYS_LOVE, "    • Ocultar Boys Love")
+        val harem = checkBox(ctx, NSFW_HAREM, "    • Ocultar Harem")
+        val trap = checkBox(ctx, NSFW_TRAP, "    • Ocultar Trap")
+
+        val nsfwPrefs = listOf(ecchi, gl, bl, harem, trap)
```

Comment:
I don't have a specific suggestion, but I think this data structure here is a bit lacking. There is no direct link between the option name and the genre ID.